util/pace/util/grid/generation.py

```diff
@@ -581,62 +581,70 @@ def dyc(self) -> util.Quantity:
         """
         if self._ks is None:
-            (
-                self._ks,
-                self._ptop,
-                self._ak,
-                self._bk,
-            ) = self._set_hybrid_pressure_coefficients()
+            if not eta_file == "None":
+                (
+                    self._ks,
+                    self._ptop,
+                    self._ak,
+                    self._bk,
+                ) = self._set_hybrid_pressure_coefficients(eta_file)
+            else:
+                raise IOError("eta file is not specified")
         return self._ks
 
-    @property
-    def ak(self) -> util.Quantity:
+    def ak(self, eta_file="None") -> util.Quantity:
         """
         the ak coefficient used to calculate the pressure at a given k-level:
         pk = ak + (bk * ps)
         """
         if self._ak is None:
-            (
-                self._ks,
-                self._ptop,
-                self._ak,
-                self._bk,
-            ) = self._set_hybrid_pressure_coefficients()
+            if not eta_file == "None":
+                (
+                    self._ks,
+                    self._ptop,
+                    self._ak,
+                    self._bk,
+                ) = self._set_hybrid_pressure_coefficients(eta_file)
+            else:
+                raise IOError("eta file is not specified")
         return self._ak
 
-    @property
-    def bk(self) -> util.Quantity:
+    def bk(self, eta_file="None") -> util.Quantity:
         """
         the bk coefficient used to calculate the pressure at a given k-level:
         pk = ak + (bk * ps)
         """
         if self._bk is None:
-            (
-                self._ks,
-                self._ptop,
-                self._ak,
-                self._bk,
-            ) = self._set_hybrid_pressure_coefficients()
+            if not eta_file == "None":
+                (
+                    self._ks,
+                    self._ptop,
+                    self._ak,
+                    self._bk,
+                ) = self._set_hybrid_pressure_coefficients(eta_file)
+            else:
+                raise IOError("eta file is not specified")
         return self._bk
 
-    @property
-    def ptop(self) -> util.Quantity:
+    def ptop(self, eta_file="None") -> util.Quantity:
         """
         the pressure of the top of atmosphere level
         """
         if self._ptop is None:
-            (
-                self._ks,
-                self._ptop,
-                self._ak,
-                self._bk,
-            ) = self._set_hybrid_pressure_coefficients()
+            if not eta_file == "None":
+                (
+                    self._ks,
+                    self._ptop,
+                    self._ak,
+                    self._bk,
+                ) = self._set_hybrid_pressure_coefficients(eta_file)
+            else:
+                raise IOError("eta file is not specified")
```

Comment:
We still want `ak`, `bk`, `ks`, and `ptop` to be properties, but we can't pass an eta file when we reference them. Loading in ak and bk values from an external file eliminates the lazy evaluation of these variables, so instead the `__init__` method should take `eta_file` as an argument and set `self._ak`, `self._bk`, `self._ptop` and `self._ks` using `set_hybrid_pressure_coefficients` and then the `ak`, `bk`, `ptop`, and `ks` properties can return the values cast to the correct precision.